### PR TITLE
Handle fileopen if running (second run)

### PR DESCRIPTION
**Pull request by HofiOne**
## Description

Added the following features:
- accept media file open requests and replace the current playlist with a new one even if the app already started
- two new Finder service items, "Enqueue next in Kodi", "Enqueue last in Kodi" to modify the current playlists without replacing it
- ability to accept file drag and drop onto application icon
- listing Kodi in "Open with..." Finder context menu for common media file types

## Motivation and Context

Kodi Mac does not allow modifying or replacing the current playlist once the app has started.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Rechi** [comment]
@HofiOne please update your changes to fit our coding style.
Especially take a look at brackets (https://codedocs.xyz/xbmc/xbmc/code_guidelines.html#code_guidelines_2).

**Comment by Rechi** [comment]
@HofiOne rebase on xbmc/master if you want the latest changes from it in your feature branch, but never ever merge master into a feature branch.

**Comment by HofiOne** [comment]
@Rechi about the coding style, in most of the touched files (and others too i saw earlier) there are mixed styles already, even within the files and more often between different files, what style to follow this way?

**Comment by Rechi** [comment]
It doesn't matter what styles are currently used in the files, the correct coding style is described at https://codedocs.xyz/xbmc/xbmc/code_guidelines.html#code_guidelines_2.
Only exception is, changing a few lines in an already existing function, then you can keep the original indentation to keep it readable.

File: xbmc/platform/darwin/osx/SDLMain.mm [hunk 1]
```diff
@@ -353,54 +374,142 @@ - (void) applicationDidFinishLaunching: (NSNotification *) note
...
+          for (NSTextCheckingResult* nextMatch in matches)
+          {
+            NSString* substringForMatch = [string substringWithRange:nextMatch.range];
+            
+            if (substringForMatch)
+            {
+              NSURL* foundURL = [NSURL URLWithString:substringForMatch];
+              NSString* fileName = nil;
+              
+              if (foundURL.isFileReferenceURL || [foundURL.scheme caseInsensitiveCompare:@"file"] == NSOrderedSame)   // Already a file system path URL
```
Comment (Memphiz) [inline xbmc/platform/darwin/osx/SDLMain.mm]: This code is the same as in the else branch or am I blind?

**Comment by Memphiz** [comment]
I would like to have this as at least 4 separated commits.

1. Info plist change for accepting the media 
2. Info plist change for added menu entries
3. that free after SDl Main
4. the whole enqueue features 

In case you spot some sane possibility to split the 4th commit even more this would be welcome aswell- but not mandatory.

Beside that - nice work man :)

File: xbmc/platform/darwin/osx/SDLMain.mm [hunk 2]
```diff
@@ -353,54 +374,142 @@ - (void) applicationDidFinishLaunching: (NSNotification *) note
...
+                fileName = foundURL.path;
+              else
+              {
+                foundURL = [NSURL fileURLWithPath:substringForMatch];
```
Comment (HofiOne) [inline xbmc/platform/darwin/osx/SDLMain.mm]: It's been a decade i've added this and at the first sight even i could not spot the difference
this is the difference, first trying to look for a match using URLWithString: second it uses fileURLWithPath:

**Comment by phunkyfish** [comment]
PR closed as it appears to be abandoned by the author. 

If the author would like it kept open please just request so on the PR. If you feel there is still merit in merging this PR you can request the label be changed to `PR Cleanup: Backburner`.

**Pull request closed without merging**